src/action.rs

```diff
@@ -186,6 +186,23 @@ where
         /// To deactivate the functionality, set this to 0.
         tap_hold_interval: u16,
     },
+    /// One-shot key. Activates `action` until a single other key is pressed following the
+    /// one shot activation or `timeout` ticks elapse. For example, a one-shot key can be
+    /// used to activate shift for exactly one keypress or switch to another layer for
+    /// exactly one keypress.
+    ///
+    /// Holding a one-shot key will be treated as a normal held keypress.
+    ///
+    /// If you use one-shot outside of its intended use cases (modifier key action or layer
+    /// action) then you will likely have undesired behaviour. E.g. one-shot with the space
+    /// key will hold space until either another key is pressed or the timeout occurs, which will
+    /// probably send many undesired space characters to your active application.
```

Comment:
what about multiple one shot? layer one shot? That's important feature IMHO. Allowing something like:
* one shot alt
* one shot ctrl
* one shot action layer
* suppr (on the action layer)
would be very soon needed.

Reply:
Yeah, multiple one shot.  Good for layer (so it works for any action, but might do strange things) .